SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -230,7 +238,12 @@ void USpatialWorkerConnection::ProcessOutgoingMessages()
 		TUniquePtr<FOutgoingMessage> OutgoingMessage;
 		OutgoingMessagesQueue.Dequeue(OutgoingMessage);
 
-		OnDequeueMessage.Broadcast(OutgoingMessage.Get());
+#if TRACE_LIB_ACTIVE
+		if (Tracer.IsValid())
+		{
+			Tracer->OnDequeueMessage(OutgoingMessage.Get());
```

Comment:
Yeah, this was more to wrap the Enqueue message which can get called from a different thread. This was removed because it seemed unlikely the delegate was much use beyond the tracer anyway, and to keep them both consistent.

Reply:
It stops the connection having to be coupled to the tracer data and stops us having to put yet more `#if` everywhere in code.

Reply:
If enqueue is called in a different thread than that in itself implies there is a race in the tracing that won't be fixed by this (and will cause missing traces on startup)

Reply:
We have a mutex in the latency tracer to prevent that.
This is to prevent UObject's and their delegates being called from multiple threads

Reply:
We have a mutex in the latency tracer to prevent that.
This is to prevent UObject's and their delegates being called from multiple threads